### PR TITLE
Safety-checker: support for SLH primitives

### LINKED ISSUE #670: Safety checker & SLH
The safety checker is not ready to handle programs using SLH primitives, e.g.,

~~~
export
fn main() {
  reg u64 ms;
  stack u8[8] s;
  reg ptr u8[8] p;
  p = s;
  ms = #init_msf();
  p = #protect_ptr(p, ms);
}
~~~

https://github.com/jasmin-lang/jasmin/blob/504d05f25c561f117b3ee2a2b664f8b692130d6c/compiler/safetylib/safetyAbsExpr.ml#L1243

### PR DESCRIPTION
**Pull request by vbgl**
Fixes #670

### CODE REVIEW

File: compiler/safetylib/safetyInterpreter.ml [hunk 1]
```diff
@@ -1352,6 +1352,19 @@ end = struct
...
 
+    | Sopn.Oslh op ->
+       begin match op with
+       | SLHinit -> [ Some (pcast U64 (Pconst (Z.of_int 0))) ]
+       | SLHupdate ->
+          let b, msf = as_seq2 es in
+          let msf = Pif (Bty (U U64), b, msf, pcast U64 (Pconst (Z.of_int (-1)))) in
+          [ Some msf ]
+       | SLHmove -> let msf = as_seq1 es in [ Some msf ]
+       | SLHprotect _ | SLHprotect_ptr _ ->
```
Comment (bgregoir) [inline compiler/safetylib/safetyInterpreter.ml:1363]: This is suspicious, what about protect(x, -1)? I think you assume that the program is well typed for SCT.

Comment (vbgl) [reply compiler/safetylib/safetyInterpreter.ml:1363]: What’s the semantics of `#protect`?

Comment (bgregoir) [reply compiler/safetylib/safetyInterpreter.ml:1363]: You are right, I forgot that the semantic of protect is the identity at source level.
Sorry for that.

**Review by bgregoir — APPROVED** [review]

File: compiler/safetylib/safetyInterpreter.ml (hunk 1 above)
Comment (vbgl) [reply compiler/safetylib/safetyInterpreter.ml:1363]: This program is well-typed and has a well-defined semantics (the identity). However, it cannot be compiled.

~~~
export
fn p(reg u64 x) -> reg u64 {
reg u64 y msf;
msf = 42;
y = #protect(x, msf);
return y;
}
~~~

**Pull request merged**